sympy/matrices/expressions/tests/test_matexpr.py

```diff
@@ -587,8 +584,8 @@ def test_MatAdd_postprocessor():
     assert a.args == (S.ComplexInfinity, z)
 
     a = Add(z, S.NaN)
-    # assert a == Add(S.NaN, z) # See the XFAIL above
-    assert isinstance(a, Add)
+    assert a == Add(S.NaN, z)
+    assert a.func == Add
```

Comment:
We can use `assert unchanged(Add, S.NaN, z)` I think

Reply:
Fixed it.